crates/provider/src/builder.rs

```diff
@@ -398,6 +395,13 @@ impl<L, F, N> ProviderBuilder<L, F, N> {
         let client = ClientBuilder::default().hyper_http(url);
         self.on_client(client)
     }
+
+    /// Aggregate multiple `eth_call` requests into a single batch request using Multicall3.
+    ///
+    /// See [`BatchLayer`] for more information.
+    pub fn with_call_batching(self) -> ProviderBuilder<Stack<BatchLayer, L>, F, N> {
+        self.layer(BatchLayer::new())
+    }
```

Comment:
this makes sense, if the user wants to configure the layer they can just call layer with the configured batchlayer manually.

Reply:
Added example to the layer docs